modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -797,6 +797,50 @@ GAPI_FLUID_KERNEL(GFluidDivRC, cv::gapi::core::GDivRC, false)
     }
 };
 
+//-------------------
+//
+// Fluid kernels: mask
+//
+//-------------------
+
+template<typename DST, typename SRC>
+static void run_mask(Buffer &dst, const View &src, const View &mask)
+{
+    static_assert(std::is_same<DST, SRC>::value,
+        "Input and output types must match");
+
+    int length  = dst.length(); // dst, src and mask have the same size and are single-channel
```

Comment:
Hmm, so we can't apply a grayscale mask to a BGR image?

Reply:
It is correct. There is a restriction in our documentation that the `mask()` takes only 1-channeled input and mask

Reply:
Ok thanks